public/content/de/basics/basic-types.md

```diff
@@ -0,0 +1,118 @@
+Ein `char` ist ein UTF-8 Zeichen, `wchar` ein UTF-16 Zeichen and `dchar`
+ein UTF-32 Zeichen.
+
+Eine Umwandlung zwischen Variablen ist nur erlaubt wenn keine Präzision verloren
+gehen kann, obgleich die Umwandlung von Gleitkommatypen (z. B. `double` zu `float`)
+erlaubt ist.
+
+Eine Umwandlung zu einem anderen Typ kann durch den Ausdruck
+`cast(TYPE) <varName>` erzwungen werden. Dieser Ausdruck sollte aber mit großer
+Vorsicht verwenden werden, da mit `cast` Ausdrücke das Typsystem ausgeschaltet
+wird.
+
+Das spezielle Schlüsselwort `auto` erzeugt eine Variable und inferiert ihren Typ
+automatisch an Hand des Ausdruck of the rechten Seite. Zum Beispiel wird
+`auto i = 7` den Typ `int` deduzieren. Beachte, dass `auto` Typen trotzdem
+statisch während der Kompilerphase festgelegt werden und nicht den Typ ändern können
+- wie für jede andere Variable mit explizitem Typ.
+
+### Eigenschaften eines Types
+
+Alle Datentypen haben die Eigenschaft `.init`, welche dem Initialwert gleicht.
+Für alle Ganzzahlen ist dies `0` und für Gleitkommazahlen ist es `nan` (vom Englischen
+"not a number"). Sowohl Ganzzahl als auch Gleitkommatypen haben eine `.min` und
+`.max` Eigenschaft für die kleinste und größte Zahl, die der Typ darstellen kann.
+Gleitkommazahlen besitzen noch weitere Eigenschaften: `nan` (der invalide Zustand),
+`infinity` (unendlich), `.dig` (Anzahl der dezimalen Stellen der Präzision) und
+`.mant_dig` (Anzahl an Bits der Mantisse) und Weitere.
+
+Zusätzlich besitzt jeder Typ die `.stringof` Eigenschaft welche seinen Namen als
+String zurückgibt.
+
+### Indizes in D
+
+In D haben Indizes den Aliasnamen `size_t` welcher groß genug ist um den alle
+Speicherzellen adressieren zu können, d.h. für ein 32-bit System `uint` und für
+ein 64-bit System `ulong`.
+
+`assert` ist eine eingebaute Anweisung welche Ausdrücke im Testmodus validiert
+und mit einen `AssertionError` wirft, wenn die Evaluation des Ausdruck `false`
+liefert.
+
+### In der Tiefe
+
+#### Grundlegende Referenzen
+
+- [Anweisungen](http://ddili.org/ders/d.en/assignment.html)
+- [Variablen](http://ddili.org/ders/d.en/variables.html)
+- [Arithmetik](http://ddili.org/ders/d.en/arithmetic.html)
+- [Gleitkommazahlen](http://ddili.org/ders/d.en/floating_point.html)
+- [Grundlegende Typen in _Programmieren in D_](http://ddili.org/ders/d.en/types.html)
+
+#### Weitergehende Referenzen
+
+- [Übersicht über alle elementaren Datentypen in D](https://dlang.org/spec/type.html)
+- [`auto` und `typeof` in _Programmieren in D_](http://ddili.org/ders/d.en/auto_and_typeof.html)
+- [Eigenschaften eines Types](https://dlang.org/spec/property.html)
+
+## {SourceCode}
+
+```d
+import std.stdio;
+
+void main()
+{
+    // Zahlen können zur Verbesserung
+    // der Lesbarkeit mit einem
+    // Unterstrich '_' separariert werden
+    int b = 7_000_000;
+    short c = cast(short) b; // cast benötigt
+    uint d = b; // größerer Typ
+    int g;
+    assert(g == 0);
+
+    auto f = 3.1415f; // .f steht für float
```

Comment:
> .f? only f

Do you want to fix the original, too?

http://tour.dlang.org/tour/en/basics/basic-types